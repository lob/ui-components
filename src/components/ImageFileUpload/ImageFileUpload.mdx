import { Canvas, ArgTypes, PRIMARY_STORY } from '@storybook/addon-docs';
import { Default, FileSizeError } from './ImageFileUpload.stories';

# Image File Upload

An image file upload component with image preview and drag and drop.

## Default Configuration

By default, the component will occupy the full height and width of its container and has a minimum height of 50px.
When an image file is selected, the image preview and remove button are positioned at the very left and right of the
component, respectively.

Note that files cannot be dropped directly onto the choose button and if the component is not wide enough, the drop zone
will be obscured by the choose button.

<Canvas of={Default} layout={'padded'} />

## Max File Size Error

Choose any image file larger than 1 byte to see the error state when the file size exceeds the configured `maxFileSize`.

<Canvas of={FileSizeError} layout={'padded'} />

## Props

<ArgTypes story={PRIMARY_STORY} />
