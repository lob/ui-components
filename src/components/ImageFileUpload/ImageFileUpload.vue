<template>
  <FileUpload
    mode="advanced"
    :file-limit="1"
    :accept="accept"
    :max-file-size="maxFileSize"
    :choose-label="chooseFileLabel"
    :pt="{
      root: function ({ props, state }) {
        return {
          class: [
            'uic-img-upload',
            {
              'uic-img-upload--basic': props.mode === 'basic',
              'uic-img-upload--advanced': props.mode === 'advanced',
              'uic-img-upload--preview': state.files?.length > 0,
              'uic-img-upload--with-messages': state.messages?.length > 0
            }
          ]
        };
      },
      buttonbar: function ({ state }) {
        return {
          class: [
            'uic-img-upload__btnbar',
            {
              'uic-img-upload__btnbar--preview': state.files?.length > 0
            }
          ]
        };
      },
      label: function () {
        return {
          class: ['uic-img-upload__label']
        };
      },
      content: function () {
        return {
          class: ['uic-img-upload__content'],
          'data-testid': 'img-upload-dropzone'
        };
      },
      fileName: function () {
        return {
          class: ['uic-img-upload__filename']
        };
      },
      fileSize: function () {
        return {
          class: ['uic-img-upload__filesize']
        };
      },
      chooseButton: function ({ props, state }) {
        return {
          class: ['uic-img-upload__choose-btn'],
          'data-testid': 'img-upload-choose-btn'
        };
      },
      chooseButtonLabel: function () {
        return {
          class: ['uic-img-upload__choose-btn-label']
        };
      },
      cancelButton: function () {
        return {
          root({ state }: FileUploadPassThroughMethodOptions) {
            return {
              class: ['uic-img-upload__cancel-btn']
            };
          },
          label: {
            class: ['uic-img-upload__label', 'sr-only']
          }
        };
      },
      uploadButton: function () {
        return {
          root() {
            return {
              class: ['uic-img-upload__upload-btn']
            };
          }
        };
      },
      removeButton: function () {
        return {
          root() {
            return {
              class: ['uic-img-upload__remove-btn']
            };
          }
        };
      },
      badge: function () {
        return {
          root() {
            return {
              class: ['uic-img-upload__badge']
            };
          }
        };
      },
      empty: function () {
        return {
          class: ['uic-img-upload__empty']
        };
      },
      file: function () {
        return {
          class: ['uic-img-upload__file']
        };
      },
      message: function () {
        return {
          root(
            _: MessagePassThroughMethodOptions<FileUploadPassThroughMethodOptions>
          ) {
            return {
              class: ['uic-img-upload__message']
            };
          },
          icon() {
            return {
              class: ['uic-img-upload__message-icon']
            };
          },
          closeButton() {
            return {
              class: ['uic-img-upload__message-close-btn']
            };
          }
        };
      }
    }"
    @select="($event) => onSelectFile($event)"
  >
    <template #empty>
      <p v-if="dropZone" class="uic-img-upload__empty-label"></p>
    </template>
    <template #chooseicon>
      <i class="pi pi-upload"></i>
    </template>
    <template #cancelicon>
      <i class="pi pi-trash"></i>
    </template>
  </FileUpload>
</template>

<script setup lang="ts">
import FileUpload, { FileUploadSelectEvent } from 'primevue/fileupload';
// eslint-disable-next-line no-unused-vars
import type { FileUploadPassThroughMethodOptions } from 'primevue/fileupload';
// eslint-disable-next-line no-unused-vars
import type { MessagePassThroughMethodOptions } from 'primevue/message';

export interface ImageFileUploadProps {
  /**
   * The name of the file input.
   */
  name: string;
  /**
   * The maximum file size in bytes.
   */
  maxFileSize?: number;
  /**
   * The accepted file types. Defaults to `image/*`.
   *
   * @see https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/file#unique_file_type_specifiers
   */
  accept?: string;
  /**
   * Indicates whether to show the drop zone. Defaults to `true`.
   */
  dropZone?: boolean;
  /**
   * The message to show when there are no selected files.
   */
  emptyLabel?: string;
  /**
   * The label for the choose file button. Defaults to `Upload`.
   */
  chooseFileLabel?: string;
}

export interface FileSelectedEvent {
  originalEvent: Event;
  file?: File;
}

/* eslint-disable no-unused-vars */
export interface ImageFileEmits {
  /**
   * Emitted when a file is selected.
   *
   * | Event Property  | Type    | Description |
   * | --------------- | ------- | ----------- |
   * | originalEvent   | Event   | The original event. |
   * | file             | File / `undefined`     | The selected file or `undefined` if the selected file exceeds the maxFileSize or is an invalid file type. |
   *
   * @event fileSelected
   */
  (e: 'fileSelected', $event: FileSelectedEvent): void;
}
/* eslint-enable no-unused-vars */

withDefaults(defineProps<ImageFileUploadProps>(), {
  maxFileSize: 1000000,
  dropZone: true,
  accept: 'image/*',
  chooseFileLabel: 'Upload',
  emptyLabel: ''
});

const emit = defineEmits<ImageFileEmits>();

function onSelectFile($event: FileUploadSelectEvent) {
  // only allow one file to be selected at a time
  if (Array.isArray($event.files) && $event.files.length > 1) {
    // keep the most recently added file and discard the rest
    $event.files.splice(0, $event.files.length - 1);
  }

  const [selectedFile] = $event.files;

  emit('fileSelected', {
    originalEvent: $event.originalEvent,
    file: selectedFile
  });
}
</script>

<style lang="scss">
.uic-img-upload {
  $root: &;

  @apply relative w-full h-full min-h-[50px] min-w-[160px];

  &--preview {
    @apply flex flex-row-reverse justify-between;

    #{$root}__choose-btn {
      @apply hidden;
    }
  }

  &--with-messages {
    #{$root}__choose-btn {
      @apply hidden;
    }

    #{$root}__content {
      @apply flex flex-col items-center;
    }
  }

  & > input[type='file'] {
    @apply hidden;
  }

  &--basic {
    input[type='file'] {
      @apply hidden;
    }
  }

  &__content {
    @apply absolute top-0 left-0 w-full h-full;
  }

  &__btnbar {
    @apply relative w-[80px] m-auto z-10;

    &--preview {
      @apply mr-0 w-fit;
    }
  }

  &__choose-btn {
    @apply cursor-pointer flex flex-col items-center justify-center gap-2;
  }

  &__choose-btn-label {
    @apply text-purple-800;
  }

  &__cancel-btn {
    &[disabled] {
      @apply hidden;
    }

    span {
      @apply sr-only;
    }
  }

  &__upload-btn {
    @apply hidden;
  }

  &__remove-btn {
    @apply hidden;
  }

  &__filename {
    @apply hidden;
  }

  &__filesize {
    @apply hidden;
  }

  &__badge {
    @apply hidden;
  }

  &__message {
    @apply flex justify-center w-full;

    & > div {
      @apply flex flex-row-reverse gap-2;
    }

    #{$root}__message-icon {
      @apply my-auto text-red-500;
    }
  }
}
</style>
