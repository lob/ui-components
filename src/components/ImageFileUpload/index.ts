export { default as ImageFileUpload } from './ImageFileUpload.vue';
