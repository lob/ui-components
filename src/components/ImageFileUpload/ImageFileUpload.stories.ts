import type { Meta, StoryObj } from '@storybook/vue3';
import ImageFileUpload from './ImageFileUpload.vue';
import mdx from './ImageFileUpload.mdx';

const meta: Meta<typeof ImageFileUpload> = {
  title: 'Components/Image File Upload',
  component: ImageFileUpload,
  parameters: {
    docs: {
      page: mdx
    }
  }
};

export default meta;

type Story = StoryObj<typeof ImageFileUpload>;

export const Default: Story = {
  args: {
    name: 'demo'
  }
};

export const FileSizeError: Story = {
  args: {
    name: 'demo-filesize-error',
    maxFileSize: 1
  }
};
