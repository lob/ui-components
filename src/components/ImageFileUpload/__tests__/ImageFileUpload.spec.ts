import '@testing-library/jest-dom';
import {
  RenderOptions,
  render,
  waitFor,
  fireEvent
} from '@testing-library/vue';
import userEvent from '@testing-library/user-event';
import ImageFileUpload from '../ImageFileUpload.vue';
import PrimeVue from 'primevue/config';
import { FileSelectedEvent } from '@/components/ImageFileUpload/ImageFileUpload.vue';

const initialProps = {
  name: 'img-file'
};

const renderComponent = (options: RenderOptions = {}) =>
  render(ImageFileUpload, {
    global: {
      plugins: [PrimeVue]
    },
    ...options
  });

describe('ImageFileUpload', () => {
  it('should render correctly', async () => {
    const { baseElement, getByTestId } = renderComponent({
      props: initialProps
    });

    // no accessibility selectors for choose button.  Screen readers will interact
    // with the input element directly.
    const chooseButton = getByTestId('img-upload-choose-btn');
    expect(chooseButton).toBeInTheDocument();
    expect(chooseButton).toBeVisible();
    // expect(baseElement).toMatchSnapshot();
  });

  it('should render the preview when a file is selected', async () => {
    const { container } = renderComponent({ props: initialProps });

    const fileInput = container.querySelector('input[type="file"]');

    expect(fileInput).toBeInTheDocument();

    const file = new File(['(⌐□_□)'], 'chucknorris.png', {
      type: 'image/png'
    });

    userEvent.upload(fileInput as HTMLElement, file);

    await waitFor(() => {
      const previewImg = container.querySelector('img');

      expect(previewImg).toBeInTheDocument();
      expect(previewImg).toHaveAttribute(
        'src',
        expect.stringMatching('^blob:')
      );
    });
  });

  it('should return to empty state after clicking the cancel button', async () => {
    const { container, getByRole, getByTestId } = renderComponent({
      props: initialProps
    });

    const fileInput = container.querySelector('input[type="file"]');

    expect(fileInput).toBeInTheDocument();

    const file = new File(['(⌐□_□)'], 'chucknorris.png', {
      type: 'image/png'
    });

    userEvent.upload(fileInput as HTMLElement, file);

    await waitFor(() => {
      const previewImg = container.querySelector('img');

      expect(previewImg).toBeInTheDocument();
      expect(previewImg).toHaveAttribute(
        'src',
        expect.stringMatching('^blob:')
      );
    });

    const cancelButton = getByRole('button', { name: 'Cancel' });

    expect(cancelButton).toBeInTheDocument();

    userEvent.click(cancelButton);

    await waitFor(() => {
      const previewImg = container.querySelector('img');
      expect(previewImg).not.toBeInTheDocument();
    });

    await waitFor(() => {
      const chooseButton = getByTestId('img-upload-choose-btn');
      expect(chooseButton).toBeInTheDocument();
    });

    await waitFor(() => {
      expect(fileInput).toHaveValue('');
    });
  });

  it('should render the image preview when a file is dropped on the dropzone', async () => {
    const { container, getByTestId } = renderComponent({ props: initialProps });

    const dropzone = getByTestId('img-upload-dropzone');

    expect(dropzone).toBeInTheDocument();

    const file = new File(['(⌐□_□)'], 'chucknorris.png', {
      type: 'image/png'
    });

    await fireEvent.drop(dropzone as HTMLElement, {
      dataTransfer: {
        files: [file]
      }
    });

    await waitFor(() => {
      const previewImg = container.querySelector('img');

      expect(previewImg).toBeInTheDocument();
      expect(previewImg).toHaveAttribute(
        'src',
        expect.stringMatching('^blob:')
      );
    });
  });

  it('should display an error when the file size exceeds the maxFileSize prop', async () => {
    const { baseElement, getByRole, container, emitted } = renderComponent({
      props: { ...initialProps, maxFileSize: 1 }
    });

    const fileInput = container.querySelector('input[type="file"]');

    expect(fileInput).toBeInTheDocument();

    const file = new File(['(⌐□_□)'], 'chucknorris.png', {
      type: 'image/png'
    });

    userEvent.upload(fileInput as HTMLElement, file);

    await waitFor(() => {
      const error = getByRole('alert');
      expect(error).toBeInTheDocument();
      expect(error).toHaveTextContent(
        'Invalid file size, file size should be smaller than 1 B'
      );
      expect(emitted()).toHaveProperty('fileSelected');
      const [fileSelectedEvent] = emitted().fileSelected as [FileSelectedEvent];
      expect(fileSelectedEvent.file).toBeUndefined();
      // expect(baseElement).toMatchSnapshot();
    });
  });

  it('should emit the fileSelected event when a file is selected', async () => {
    const { container, emitted } = renderComponent({ props: initialProps });

    const fileInput = container.querySelector('input[type="file"]');

    expect(fileInput).toBeInTheDocument();

    const file = new File(['(⌐□_□)'], 'chucknorris.png', {
      type: 'image/png'
    });

    userEvent.upload(fileInput as HTMLElement, file);

    await waitFor(() => {
      expect(emitted()).toHaveProperty('fileSelected');
      const [fileSelectedEvent] = emitted().fileSelected[0] as [
        FileSelectedEvent
      ];

      expect(fileSelectedEvent.file).toBeInstanceOf(File);
      expect(fileSelectedEvent.file?.name).toBe('chucknorris.png');
      expect(fileSelectedEvent.originalEvent).toBeInstanceOf(Event);
    });
  });
});
