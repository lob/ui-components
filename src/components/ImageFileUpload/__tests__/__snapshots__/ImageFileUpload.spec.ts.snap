// Vitest Snapshot v1

exports[`ImageFileUpload > should display an error when the file size exceeds the maxFileSize prop 1`] = `
<body>
  <div>
    <div
      class="p-fileupload p-fileupload-advanced p-component uic-img-upload uic-img-upload--advanced uic-img-upload--with-messages"
      data-pc-name="fileupload"
      data-pc-section="root"
    >
      <input
        accept="image/*"
        data-pc-section="input"
        type="file"
      />
      <div
        class="p-fileupload-buttonbar uic-img-upload__btnbar"
        data-pc-section="buttonbar"
      >
        
        <span
          class="p-button p-component p-fileupload-choose uic-img-upload__choose-btn"
          data-pc-section="choosebutton"
          data-testid="img-upload-choose-btn"
          tabindex="0"
        >
          
          <i
            class="pi pi-upload"
          />
          
          <span
            class="p-button-label uic-img-upload__choose-btn-label"
            data-pc-section="choosebuttonlabel"
          >
            Upload
          </span>
        </span>
        <button
          aria-label="Upload"
          class="p-button p-component p-disabled uic-img-upload__upload-btn"
          data-pc-name="button"
          data-pc-section="uploadbutton"
          disabled=""
          type="button"
        >
          
          
          
          <svg
            aria-hidden="true"
            class="p-icon p-button-icon p-button-icon-left"
            data-pc-section="uploadbuttonicon"
            fill="none"
            height="14"
            viewBox="0 0 14 14"
            width="14"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              clipPath="url(#pv_icon_clip_pv_id_5)"
            >
              <path
                clip-rule="evenodd"
                d="M6.58942 9.82197C6.70165 9.93405 6.85328 9.99793 7.012 10C7.17071 9.99793 7.32234 9.93405 7.43458 9.82197C7.54681 9.7099 7.61079 9.55849 7.61286 9.4V2.04798L9.79204 4.22402C9.84752 4.28011 9.91365 4.32457 9.98657 4.35479C10.0595 4.38502 10.1377 4.40039 10.2167 4.40002C10.2956 4.40039 10.3738 4.38502 10.4467 4.35479C10.5197 4.32457 10.5858 4.28011 10.6413 4.22402C10.7538 4.11152 10.817 3.95902 10.817 3.80002C10.817 3.64102 10.7538 3.48852 10.6413 3.37602L7.45127 0.190618C7.44656 0.185584 7.44176 0.180622 7.43687 0.175736C7.32419 0.063214 7.17136 0 7.012 0C6.85264 0 6.69981 0.063214 6.58712 0.175736C6.58181 0.181045 6.5766 0.186443 6.5715 0.191927L3.38282 3.37602C3.27669 3.48976 3.2189 3.6402 3.22165 3.79564C3.2244 3.95108 3.28746 4.09939 3.39755 4.20932C3.50764 4.31925 3.65616 4.38222 3.81182 4.38496C3.96749 4.3877 4.11814 4.33001 4.23204 4.22402L6.41113 2.04807V9.4C6.41321 9.55849 6.47718 9.7099 6.58942 9.82197ZM11.9952 14H2.02883C1.751 13.9887 1.47813 13.9228 1.22584 13.8061C0.973545 13.6894 0.746779 13.5241 0.558517 13.3197C0.370254 13.1154 0.22419 12.876 0.128681 12.6152C0.0331723 12.3545 -0.00990605 12.0775 0.0019109 11.8V9.40005C0.0019109 9.24092 0.065216 9.08831 0.1779 8.97579C0.290584 8.86326 0.443416 8.80005 0.602775 8.80005C0.762134 8.80005 0.914966 8.86326 1.02765 8.97579C1.14033 9.08831 1.20364 9.24092 1.20364 9.40005V11.8C1.18295 12.0376 1.25463 12.274 1.40379 12.4602C1.55296 12.6463 1.76817 12.7681 2.00479 12.8H11.9952C12.2318 12.7681 12.447 12.6463 12.5962 12.4602C12.7453 12.274 12.817 12.0376 12.7963 11.8V9.40005C12.7963 9.24092 12.8596 9.08831 12.9723 8.97579C13.085 8.86326 13.2378 8.80005 13.3972 8.80005C13.5565 8.80005 13.7094 8.86326 13.8221 8.97579C13.9347 9.08831 13.998 9.24092 13.998 9.40005V11.8C14.022 12.3563 13.8251 12.8996 13.45 13.3116C13.0749 13.7236 12.552 13.971 11.9952 14Z"
                fill="currentColor"
                fill-rule="evenodd"
              />
            </g>
            <defs>
              <clippath
                id="pv_icon_clip_pv_id_5"
              >
                <rect
                  fill="white"
                  height="14"
                  width="14"
                />
              </clippath>
            </defs>
          </svg>
          
          
          <span
            class="p-button-label"
            data-pc-section="label"
          >
            Upload
          </span>
          <!---->
          
        </button>
        <button
          aria-label="Cancel"
          class="p-button p-component p-disabled uic-img-upload__cancel-btn"
          data-pc-name="button"
          data-pc-section="cancelbutton"
          disabled=""
          type="button"
        >
          
          
          
          <i
            class="pi pi-trash"
          />
          
          
          <span
            class="p-button-label uic-img-upload__label sr-only"
            data-pc-section="label"
          >
            Cancel
          </span>
          <!---->
          
        </button>
        
      </div>
      <div
        class="p-fileupload-content uic-img-upload__content"
        data-p-highlight="false"
        data-pc-section="content"
        data-testid="img-upload-dropzone"
      >
        
        <!---->
        
        <transition-stub>
          <div
            aria-atomic="true"
            aria-live="assertive"
            class="p-message p-component p-message-error uic-img-upload__message"
            data-pc-name="message"
            data-pc-section="root"
            role="alert"
          >
            <div
              class="p-message-wrapper"
              data-pc-section="wrapper"
            >
              
              <svg
                aria-hidden="true"
                class="p-icon p-message-icon uic-img-upload__message-icon"
                data-pc-section="icon"
                fill="none"
                height="14"
                viewBox="0 0 14 14"
                width="14"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  clipPath="url(#pv_icon_clip_pv_id_6)"
                >
                  <path
                    clip-rule="evenodd"
                    d="M7 14C5.61553 14 4.26215 13.5895 3.11101 12.8203C1.95987 12.0511 1.06266 10.9579 0.532846 9.67879C0.00303296 8.3997 -0.13559 6.99224 0.134506 5.63437C0.404603 4.2765 1.07129 3.02922 2.05026 2.05026C3.02922 1.07129 4.2765 0.404603 5.63437 0.134506C6.99224 -0.13559 8.3997 0.00303296 9.67879 0.532846C10.9579 1.06266 12.0511 1.95987 12.8203 3.11101C13.5895 4.26215 14 5.61553 14 7C14 8.85652 13.2625 10.637 11.9497 11.9497C10.637 13.2625 8.85652 14 7 14ZM7 1.16667C5.84628 1.16667 4.71846 1.50879 3.75918 2.14976C2.79989 2.79074 2.05222 3.70178 1.61071 4.76768C1.16919 5.83358 1.05367 7.00647 1.27876 8.13803C1.50384 9.26958 2.05941 10.309 2.87521 11.1248C3.69102 11.9406 4.73042 12.4962 5.86198 12.7212C6.99353 12.9463 8.16642 12.8308 9.23232 12.3893C10.2982 11.9478 11.2093 11.2001 11.8502 10.2408C12.4912 9.28154 12.8333 8.15373 12.8333 7C12.8333 5.45291 12.2188 3.96918 11.1248 2.87521C10.0308 1.78125 8.5471 1.16667 7 1.16667ZM4.66662 9.91668C4.58998 9.91704 4.51404 9.90209 4.44325 9.87271C4.37246 9.84333 4.30826 9.8001 4.2544 9.74557C4.14516 9.6362 4.0838 9.48793 4.0838 9.33335C4.0838 9.17876 4.14516 9.0305 4.2544 8.92113L6.17553 7L4.25443 5.07891C4.15139 4.96832 4.09529 4.82207 4.09796 4.67094C4.10063 4.51982 4.16185 4.37563 4.26872 4.26876C4.3756 4.16188 4.51979 4.10066 4.67091 4.09799C4.82204 4.09532 4.96829 4.15142 5.07887 4.25446L6.99997 6.17556L8.92106 4.25446C9.03164 4.15142 9.1779 4.09532 9.32903 4.09799C9.48015 4.10066 9.62434 4.16188 9.73121 4.26876C9.83809 4.37563 9.89931 4.51982 9.90198 4.67094C9.90464 4.82207 9.84855 4.96832 9.74551 5.07891L7.82441 7L9.74554 8.92113C9.85478 9.0305 9.91614 9.17876 9.91614 9.33335C9.91614 9.48793 9.85478 9.6362 9.74554 9.74557C9.69168 9.8001 9.62748 9.84333 9.55669 9.87271C9.4859 9.90209 9.40996 9.91704 9.33332 9.91668C9.25668 9.91704 9.18073 9.90209 9.10995 9.87271C9.03916 9.84333 8.97495 9.8001 8.9211 9.74557L6.99997 7.82444L5.07884 9.74557C5.02499 9.8001 4.96078 9.84333 4.88999 9.87271C4.81921 9.90209 4.74326 9.91704 4.66662 9.91668Z"
                    fill="currentColor"
                    fill-rule="evenodd"
                  />
                </g>
                <defs>
                  <clippath
                    id="pv_icon_clip_pv_id_6"
                  >
                    <rect
                      fill="white"
                      height="14"
                      width="14"
                    />
                  </clippath>
                </defs>
              </svg>
              
              <div
                class="p-message-text p-message-text"
                data-pc-section="text"
              >
                
                chucknorris.png: Invalid file size, file size should be smaller than 1 B.
                
              </div>
              <button
                aria-label="Close"
                class="p-message-close p-link uic-img-upload__message-close-btn"
                data-pc-section="closebutton"
                type="button"
              >
                
                <svg
                  aria-hidden="true"
                  class="p-icon p-message-close-icon"
                  data-pc-section="closeicon"
                  fill="none"
                  height="14"
                  viewBox="0 0 14 14"
                  width="14"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M8.01186 7.00933L12.27 2.75116C12.341 2.68501 12.398 2.60524 12.4375 2.51661C12.4769 2.42798 12.4982 2.3323 12.4999 2.23529C12.5016 2.13827 12.4838 2.0419 12.4474 1.95194C12.4111 1.86197 12.357 1.78024 12.2884 1.71163C12.2198 1.64302 12.138 1.58893 12.0481 1.55259C11.9581 1.51625 11.8617 1.4984 11.7647 1.50011C11.6677 1.50182 11.572 1.52306 11.4834 1.56255C11.3948 1.60204 11.315 1.65898 11.2488 1.72997L6.99067 5.98814L2.7325 1.72997C2.59553 1.60234 2.41437 1.53286 2.22718 1.53616C2.03999 1.53946 1.8614 1.61529 1.72901 1.74767C1.59663 1.88006 1.5208 2.05865 1.5175 2.24584C1.5142 2.43303 1.58368 2.61419 1.71131 2.75116L5.96948 7.00933L1.71131 11.2675C1.576 11.403 1.5 11.5866 1.5 11.7781C1.5 11.9696 1.576 12.1532 1.71131 12.2887C1.84679 12.424 2.03043 12.5 2.2219 12.5C2.41338 12.5 2.59702 12.424 2.7325 12.2887L6.99067 8.03052L11.2488 12.2887C11.3843 12.424 11.568 12.5 11.7594 12.5C11.9509 12.5 12.1346 12.424 12.27 12.2887C12.4053 12.1532 12.4813 11.9696 12.4813 11.7781C12.4813 11.5866 12.4053 11.403 12.27 11.2675L8.01186 7.00933Z"
                    fill="currentColor"
                  />
                </svg>
                
              </button>
            </div>
          </div>
        </transition-stub>
        
        <!---->
        
        
        
        <div
          class="p-fileupload-empty uic-img-upload__empty"
          data-pc-section="empty"
        >
          
          <p
            class="uic-img-upload__empty-label"
          />
          
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`ImageFileUpload > should render correctly 1`] = `
<body>
  <div>
    <div
      class="p-fileupload p-fileupload-advanced p-component uic-img-upload uic-img-upload--advanced"
      data-pc-name="fileupload"
      data-pc-section="root"
    >
      <input
        accept="image/*"
        data-pc-section="input"
        type="file"
      />
      <div
        class="p-fileupload-buttonbar uic-img-upload__btnbar"
        data-pc-section="buttonbar"
      >
        
        <span
          class="p-button p-component p-fileupload-choose uic-img-upload__choose-btn"
          data-pc-section="choosebutton"
          data-testid="img-upload-choose-btn"
          tabindex="0"
        >
          
          <i
            class="pi pi-upload"
          />
          
          <span
            class="p-button-label uic-img-upload__choose-btn-label"
            data-pc-section="choosebuttonlabel"
          >
            Upload
          </span>
        </span>
        <button
          aria-label="Upload"
          class="p-button p-component p-disabled uic-img-upload__upload-btn"
          data-pc-name="button"
          data-pc-section="uploadbutton"
          disabled=""
          type="button"
        >
          
          
          
          <svg
            aria-hidden="true"
            class="p-icon p-button-icon p-button-icon-left"
            data-pc-section="uploadbuttonicon"
            fill="none"
            height="14"
            viewBox="0 0 14 14"
            width="14"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              clipPath="url(#pv_icon_clip_pv_id_1)"
            >
              <path
                clip-rule="evenodd"
                d="M6.58942 9.82197C6.70165 9.93405 6.85328 9.99793 7.012 10C7.17071 9.99793 7.32234 9.93405 7.43458 9.82197C7.54681 9.7099 7.61079 9.55849 7.61286 9.4V2.04798L9.79204 4.22402C9.84752 4.28011 9.91365 4.32457 9.98657 4.35479C10.0595 4.38502 10.1377 4.40039 10.2167 4.40002C10.2956 4.40039 10.3738 4.38502 10.4467 4.35479C10.5197 4.32457 10.5858 4.28011 10.6413 4.22402C10.7538 4.11152 10.817 3.95902 10.817 3.80002C10.817 3.64102 10.7538 3.48852 10.6413 3.37602L7.45127 0.190618C7.44656 0.185584 7.44176 0.180622 7.43687 0.175736C7.32419 0.063214 7.17136 0 7.012 0C6.85264 0 6.69981 0.063214 6.58712 0.175736C6.58181 0.181045 6.5766 0.186443 6.5715 0.191927L3.38282 3.37602C3.27669 3.48976 3.2189 3.6402 3.22165 3.79564C3.2244 3.95108 3.28746 4.09939 3.39755 4.20932C3.50764 4.31925 3.65616 4.38222 3.81182 4.38496C3.96749 4.3877 4.11814 4.33001 4.23204 4.22402L6.41113 2.04807V9.4C6.41321 9.55849 6.47718 9.7099 6.58942 9.82197ZM11.9952 14H2.02883C1.751 13.9887 1.47813 13.9228 1.22584 13.8061C0.973545 13.6894 0.746779 13.5241 0.558517 13.3197C0.370254 13.1154 0.22419 12.876 0.128681 12.6152C0.0331723 12.3545 -0.00990605 12.0775 0.0019109 11.8V9.40005C0.0019109 9.24092 0.065216 9.08831 0.1779 8.97579C0.290584 8.86326 0.443416 8.80005 0.602775 8.80005C0.762134 8.80005 0.914966 8.86326 1.02765 8.97579C1.14033 9.08831 1.20364 9.24092 1.20364 9.40005V11.8C1.18295 12.0376 1.25463 12.274 1.40379 12.4602C1.55296 12.6463 1.76817 12.7681 2.00479 12.8H11.9952C12.2318 12.7681 12.447 12.6463 12.5962 12.4602C12.7453 12.274 12.817 12.0376 12.7963 11.8V9.40005C12.7963 9.24092 12.8596 9.08831 12.9723 8.97579C13.085 8.86326 13.2378 8.80005 13.3972 8.80005C13.5565 8.80005 13.7094 8.86326 13.8221 8.97579C13.9347 9.08831 13.998 9.24092 13.998 9.40005V11.8C14.022 12.3563 13.8251 12.8996 13.45 13.3116C13.0749 13.7236 12.552 13.971 11.9952 14Z"
                fill="currentColor"
                fill-rule="evenodd"
              />
            </g>
            <defs>
              <clippath
                id="pv_icon_clip_pv_id_1"
              >
                <rect
                  fill="white"
                  height="14"
                  width="14"
                />
              </clippath>
            </defs>
          </svg>
          
          
          <span
            class="p-button-label"
            data-pc-section="label"
          >
            Upload
          </span>
          <!---->
          
        </button>
        <button
          aria-label="Cancel"
          class="p-button p-component p-disabled uic-img-upload__cancel-btn"
          data-pc-name="button"
          data-pc-section="cancelbutton"
          disabled=""
          type="button"
        >
          
          
          
          <i
            class="pi pi-trash"
          />
          
          
          <span
            class="p-button-label uic-img-upload__label sr-only"
            data-pc-section="label"
          >
            Cancel
          </span>
          <!---->
          
        </button>
        
      </div>
      <div
        class="p-fileupload-content uic-img-upload__content"
        data-p-highlight="false"
        data-pc-section="content"
        data-testid="img-upload-dropzone"
      >
        
        <!---->
        
        
        <!---->
        
        
        
        <div
          class="p-fileupload-empty uic-img-upload__empty"
          data-pc-section="empty"
        >
          
          <p
            class="uic-img-upload__empty-label"
          />
          
        </div>
      </div>
    </div>
  </div>
</body>
`;
